src/utils/getUrlByRequestOptions.ts

```diff
@@ -159,7 +166,10 @@ export function getUrlByRequestOptions(options: ResolvedRequestOptions): URL {
     : ''
   logger.info('auth string:', authString)
 
-  const url = new URL(`${protocol}//${authString}${hostname}${path}`)
+  const url = new URL(`${protocol}//${hostname}${path}`)
+  url.username = credentials?.username || ''
```

Comment:
TIL you can set `username` and `password` on the existing URL instance since `href` is a getter. This makes the URL construct a bit more readable.